components/wso2is.key.manager.core/src/main/java/org/wso2/is/key/manager/core/tokenmgt/util/CacheProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.is.key.manager.core.tokenmgt.util;
+
+import org.wso2.is.key.manager.core.tokenmgt.handlers.ResourceConstants;
+
+import javax.cache.Cache;
+import javax.cache.Caching;
+
+/**
+ * Class for initiating and returning caches. Creating cache take place when super tenant loading and tenant loading
+ */
+public class CacheProvider {
```

Comment:
make this class singleton

Reply:
Fixed via https://github.com/wso2-extensions/apim-km-wso2is/pull/44/commits/99f0a535cea4cb836973599fccc03349753729e5